pyomo/solvers/tests/piecewise_linear/test_examples.py

```diff
@@ -39,57 +40,88 @@ def run_convert2lp(self, name):
+        _out, _log = join(currdir,'unknown.lp'), currdir+'nonconvex.lp'
+        self.assertTrue(cmp(_out, _log),
+                        msg="Files %s and %s differ" % (_out, _log))
 
     def test_nonconvex_nl(self):
         """Test examples/pyomo/piecewise/nonconvex.py"""
         self.run_convert2nl('nonconvex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'nonconvex.nl', tolerance=_NL_diff_tol)
+        _out, _log = join(currdir,'unknown.nl'), currdir+'nonconvex.nl'
+        self.assertTrue(cmp(_out, _log),
+                        msg="Files %s and %s differ" % (_out, _log))
         os.remove(join(currdir,'unknown.row'))
         os.remove(join(currdir,'unknown.col'))
 
     def test_convex_lp(self):
         """Test examples/pyomo/piecewise/convex.py"""
         self.run_convert2lp('convex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'convex.lp', tolerance=_LP_diff_tol)
+        _out, _log = join(currdir,'unknown.lp'), currdir+'convex.lp'
+        self.assertTrue(cmp(_out, _log),
+                        msg="Files %s and %s differ" % (_out, _log))
 
     def test_convex_nl(self):
         """Test examples/pyomo/piecewise/convex.py"""
         self.run_convert2nl('convex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'convex.nl', tolerance=_NL_diff_tol)
+        _out, _log = join(currdir,'unknown.nl'), currdir+'convex.nl'
+        self.assertTrue(cmp(_out, _log),
+                        msg="Files %s and %s differ" % (_out, _log))
         os.remove(join(currdir,'unknown.row'))
         os.remove(join(currdir,'unknown.col'))
 
     def test_indexed_lp(self):
         """Test examples/pyomo/piecewise/indexed.py"""
         self.run_convert2lp('indexed.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'indexed.lp', tolerance=_LP_diff_tol)
+        with open(join(currdir,'unknown.lp'), 'r') as f1, \
+            open(currdir+'indexed.lp', 'r') as f2:
+                f1_contents = list(filter(None, f1.read().split()))
+                f2_contents = list(filter(None, f2.read().split()))
+                for item1, item2 in zip(f1_contents, f2_contents):
+                    try:
+                        self.assertAlmostEqual(float(item1), float(item2))
+                    except:
+                        self.assertEqual(item1, item2)
 
     def test_indexed_nl(self):
         """Test examples/pyomo/piecewise/indexed.py"""
         self.run_convert2nl('indexed.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'indexed.nl', tolerance=_NL_diff_tol)
+        with open(join(currdir,'unknown.nl'), 'r') as f1, \
+            open(currdir+'indexed.nl', 'r') as f2:
+                f1_contents = list(filter(None, f1.read().split()))
+                f2_contents = list(filter(None, f2.read().split()))
+                for item1, item2 in zip(f1_contents, f2_contents):
```

Comment:
This should probably use `itertools.zip_longest` so that differing length files cause errors.